src/realm/group_writer.cpp

```diff
@@ -350,6 +363,27 @@ ref_type GroupWriter::write_group()
             top.set(8, value_3); // Throws
         }
     }
+    if (top.size() > Group::s_evacuation_point_ndx) {
+        ref_type ref = top.get_as_ref(Group::s_evacuation_point_ndx);
+        if (m_evacuation_limit) {
```

Comment:
Wouldn't it be simpler to just always have (and have to write) the array?

Reply:
Not sure. It would at least be more difficult to update the array after having re-created the free-lists.